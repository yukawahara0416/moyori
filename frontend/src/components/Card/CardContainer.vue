<template>
  <v-row align="stretch" class="mt-0 mb-3 ml-0 mr-3" justify="space-around">
    <v-col
      class="pb-0 pr-0"
      cols="12"
      sm="6"
      :md="md"
      :lg="lg"
      v-for="(spot, id) in spots"
      :key="id"
    >
      <card-frame :spot="spot" :id="id" />
    </v-col>

    <!-- カード配置調整（中央寄せ && 最下層端数左寄せ） -->
    <v-col
      class="ma-0 pa-0"
      cols="12"
      sm="6"
      :md="md"
      :lg="lg"
      v-for="e in empties"
      :key="e"
    >
      <card-container-fill />
    </v-col>
  </v-row>
</template>

<script>
import CardFrame from '@/components/Card/CardFrame.vue'
import CardContainerFill from '@/components/Card/CardContainerFill.vue'

export default {
  props: {
    spots: {
      type: Array,
      default: () => {
        return []
      },
      required: true
    }
  },

  components: {
    CardFrame,
    CardContainerFill
  },

  computed: {
    empties() {
      const number = this.spots.length === 1 ? 2 : this.spots.length
      return Array.from(new Array(number))
    },

    lg() {
      const route = this.$route.name

      if (route == 'profile') return 3
      return 4
    },

    md() {
      const route = this.$route.name
      if (route == 'profile') return 4
      return 6
    }
  }
}
</script>
