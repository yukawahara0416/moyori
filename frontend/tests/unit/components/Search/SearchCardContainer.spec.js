import { mount, shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import Vuetify from 'vuetify'
import Component from '@/components/Search/SearchCardContainer.vue'
import CardContainer from '@/components/Card/CardContainer.vue'

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(Vuetify)

let wrapper
let spot
let store
let vuetify

beforeEach(() => {
  spot = {
    namespaced: true,
    getters: {
      filteredSpots: () => [{ data: { id: 1 } }, { data: { id: 2 } }]
    }
  }

  store = new Vuex.Store({
    modules: {
      spot
    }
  })

  vuetify = new Vuetify()

  wrapper = shallowMount(Component, {
    localVue,
    store,
    vuetify
  })
})

describe('getters', () => {
  it('spots', () => {
    expect(wrapper.vm.spots).toEqual(store.getters.spots)
  })
})

describe('computed', () => {
  it('styleVariables', async () => {
    await wrapper.setData({ height: window.innerHeight })
    expect(wrapper.vm.styleVariables).toMatchObject({
      '--checkbox-height': `${wrapper.vm.height}px`
    })
  })
})

describe('methods', () => {
  it('handleResize', async () => {
    await wrapper.setData({ height: 100 })
    wrapper.vm.handleResize()
    expect(wrapper.vm.height).toEqual(window.innerHeight)
  })
})

describe('template', () => {
  it('v-container has :style', async () => {
    wrapper = mount(Component, {
      localVue,
      store,
      vuetify,
      stubs: ['card-container']
    })

    await wrapper.setData({ height: window.innerHeight })
    expect(wrapper.find('.container').attributes().style).toEqual(
      `--checkbox-height: ${window.innerHeight}px;`
    )
  })

  it('card-container has :spots', () => {
    expect(wrapper.find(CardContainer).props().spots).toMatchObject(
      wrapper.vm.filteredSpots
    )
  })

  it('snapshot', () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
