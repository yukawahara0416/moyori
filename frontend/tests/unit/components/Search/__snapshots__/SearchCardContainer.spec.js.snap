// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template snapshot 1`] = `
<v-col-stub
  class="pa-0 [object Object]"
  cols="12"
  md="6"
  no-gutter=""
  tag="div"
>
  <v-container-stub
    class="overflow-y-auto"
    fluid="true"
    id="scroll-target"
    tag="div"
  >
    <search-card-container-fill-stub
      spots="[object Object],[object Object]"
    />
     
    <card-container-stub
      spots="[object Object],[object Object]"
      type="map"
    />
  </v-container-stub>
</v-col-stub>
`;
